src/http/one/Tokenizer.cc

```diff
@@ -61,12 +43,14 @@ Http::One::Tokenizer::qdText(SBuf &returnedToken, const bool http1p0)
     // best we can do is a conditional reference since http1p0 value may change per-client
     const CharacterSet &tokenChars = (http1p0 ? qdtext1p0 : qdtext1p1);
 
-    for (;;) {
-        SBuf::size_type prefixLen = buf().findFirstNotOf(tokenChars);
-        returnedToken.append(consume(prefixLen));
+    while (!tok.atEnd()) {
+        if (tok.skip('"'))
```

Comment:
Move this `if` statement to the end of the loop to simplify the overall logic (with quoted string _termination_ condition checked _last_) and to optimize the handling of a common case where we can parse the whole quoted string in one loop iteration.